File: data/mods/ruby/lib/robot.rb
class Robot
  laby_name_Void = 0
  laby_name_Wall = 1
  laby_name_Rock = 2
  laby_name_Web = 3
  laby_name_Exit = 4
  laby_name_Unknown = 5

  def initialize
    perform "start"
    laby_name_ant
  end

  def output s
    puts s
    $stdout.flush
  end

  def input
    l = gets rescue ""
    if l == "quit\n" then exit(1) end;
    l
  end

  def perform action
    output action
    input
  end

  def laby_name_look
    case perform('look')
    when "void\n"
      laby_name_Void
    when "wall\n"
      laby_name_Wall
    when "rock\n"
      laby_name_Rock
    when "web\n"
      laby_name_Web
    when "exit\n"
      laby_name_Exit
    else
      laby_name_Unknown
    end
  end

  def laby_name_left; perform 'left'; end
  def laby_name_right; perform 'right'; end
  def laby_name_forward; perform 'forward'; end
  def laby_name_take; perform 'take'; end
  def laby_name_drop; perform 'drop'; end
  def laby_name_escape; perform 'escape'; end

  def laby_name_say s; perform ('say ' + s) end
end

File: data/mods/lua/lib/robot.lua
function output(s)
  io.write(s, "\n");
  io.flush();
end

function input()
    r = io.read();
    if (r == "quit") then exit(0) end;
    return r;
end

function laby_name_left()
  output("left"); input();
end

function laby_name_right()
  output("right"); input();
end

function laby_name_forward()
  output("forward"); input();
end

function laby_name_take()
  output("take"); input();
end

function laby_name_drop()
  output("drop"); input();
end

function laby_name_escape()
  output("escape"); input();
end

function laby_name_say(s)
  output("say " .. s); input();
end

laby_name_Void = 0;
laby_name_Wall = 1;
laby_name_Rock = 2;
laby_name_Web = 3;
laby_name_Exit = 4;
laby_name_Unknown = 5;

function laby_name_look()
  output("look");
  local ans = input();
  if (ans == "void") then return laby_name_Void end;
  if (ans == "wall") then return laby_name_Wall end;
  if (ans == "rock") then return laby_name_Rock end;
  if (ans == "web") then return laby_name_Web end;
  if (ans == "exit") then return laby_name_Exit end;
  return laby_name_Unknown;
end

output("start");
input()
